src/Sylius/Bundle/ApiBundle/spec/Validator/Constraints/ShopUserResetPasswordTokenNotExpiredValidatorSpec.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ApiBundle\Validator\Constraints\ShopUserResetPasswordTokenNotExpired;
+use Sylius\Component\User\Model\UserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidatorInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+
+final class ShopUserResetPasswordTokenNotExpiredValidatorSpec extends ObjectBehavior
+{
+    function let(UserRepositoryInterface $userRepository): void
+    {
+        $this->beConstructedWith($userRepository, 'P1D');
+    }
+
+    function it_is_a_constraint_validator(): void
+    {
+        $this->shouldImplement(ConstraintValidatorInterface::class);
+    }
+
+    function it_throws_an_exception_if_value_is_not_a_string(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', [null, new class() extends Constraint {
+            }])
+        ;
+    }
+
+    function it_throws_an_exception_if_constraint_is_not_a_resetPasswordTokenExpires_constraint(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', ['', new class() extends Constraint {
+            }])
+        ;
+    }
+
+    function it_does_not_add_violation_if_reset_password_token_does_not_exist(
+        UserRepositoryInterface $userRepository,
+        ExecutionContextInterface $executionContext
+    ): void {
+        $this->initialize($executionContext);
+
+        $value = 'token';
+
+        $userRepository->findOneBy(['passwordResetToken' => 'token'])->willReturn(null);
```

Comment:
```suggestion
        $userRepository->findOneBy(['passwordResetToken' => $value])->willReturn(null);
```
either that or remove the var and use string everywhere.
Pretty much the same case in other methods and the other spec